db/db.go

```diff
@@ -2,25 +2,81 @@ package db
 
 import (
 	"fmt"
+	"strings"
 
-	"github.com/getAlby/hub/db/migrations"
-	"github.com/getAlby/hub/logger"
+	"gorm.io/driver/postgres"
 	"gorm.io/driver/sqlite"
 	"gorm.io/gorm"
 	gorm_logger "gorm.io/gorm/logger"
+
+	"github.com/getAlby/hub/db/migrations"
+	"github.com/getAlby/hub/logger"
 )
 
+var SerializeTransactions = false
+
+type Config struct {
+	URI        string
+	LogQueries bool
+	DriverName string
+}
+
 func NewDB(uri string, logDBQueries bool) (*gorm.DB, error) {
+	return NewDBWithConfig(&Config{
+		URI:        uri,
+		LogQueries: logDBQueries,
+		DriverName: "",
+	})
+}
 
-	config := &gorm.Config{
+func NewDBWithConfig(cfg *Config) (*gorm.DB, error) {
+	gormConfig := &gorm.Config{
 		TranslateError: true,
 	}
-	if logDBQueries {
-		config.Logger = gorm_logger.Default.LogMode(gorm_logger.Info)
+	if cfg.LogQueries {
+		gormConfig.Logger = gorm_logger.Default.LogMode(gorm_logger.Info)
+	}
+
+	var ret *gorm.DB
+
+	if IsPostgresURI(cfg.URI) {
```

Comment:
Leaving a comment to look into the timestamp issue. I am not sure what the problem is yet.